18 creation of main view

Javi1499 opened the pull request:
# Description
What did you implemented/Why did you implemented this:
 -What? I added main view where employees will done check in/out. It incluide controller, new view and changes in model and implementation of flash messages
 - Why? Flash Messages was implemented  to return feedback to user about the action
 
  ## Screenshots
![imagen](https://user-images.githubusercontent.com/42743118/158912484-95e8dc73-ed68-4e9c-b1aa-9e10facec828.png)

femm34 committed:
Design proposal completed

We made the design for each of the views that the check clock will have

Co-authored-by: EliasUR <eliasrodriguez2002@hotmail.com>
Co-authored-by: Javi1499 <fruiz1@ucol.mx>

Javi1499 committed:
Styles loaded and input to checkin/out was added

Javi1499 committed:
Styles loaded and input to checkin/out was added

Javi1499 committed:
Merge branch 'master' into 18-creation-of-main-view

Javi1499 committed:
Feuture to do check in/out working

Javi1499 committed:
Feuture to do check in/out working

femm34 committed:
modifying .scss files

modify the application.sass.scss to import bootstrap, also modified the script build:css to compile sass

femm34 committed:
i added a table to show the branches

i used the bootstrap clases to make the table quickly and also i implemented a button to add a new branch

femm34 committed:
adding input and modifying styles

i added a input to search branches and also modified a little the table and buttons styles

femm34 committed:
added a name to input

femm34 committed:
doing the search method for branches

femm34 committed:
modifying the search method

femm34 committed:
search functionality is almost done

femm34 committed:
the search path was changed

now redirects to branches index inside of teh table

femm34 committed:
search view was removed

femm34 committed:
search path was deleted from routes.rb file

Javi1499 committed:
Flash messages were added

Javi1499 committed:
Flash messages were added

femm34 committed:
new method from index view

femm34 committed:
i used rubocop to improve syntax

femm34 committed:
Merge pull request #14 from BC-MAY-21-ROR/10-creation-of-the-design

Design proposal completed

femm34 committed:
modified controller to render on the new view

vyk2rr commented on app/controllers/records_controller.rb:
```diff
@@ -0,0 +1,31 @@
+  end
+
+  def create
+    id_employee = Employee.select(:id).where(private_number: params['private_number']).first
+    puts id_employee
+    time_range = (Time.now.midnight)..(Time.now.midnight + 1.day - 1.second)
+    record = Record.joins(:employee).where(employee: { id: id_employee.id }, check_out: nil,
+                                           created_at: time_range)
+    time = Time.now
+    if record.length.positive?
+      Record.update(record[0].id, check_out: time, hours: helpers.deff_hours(record[0].check_in, time))
+      flash[:success] = "CheckOut at #{time.to_formatted_s(:short)}. Have a good night "
+    else
+      Record.create! employee_id: id_employee.id, check_in: time
+      flash[:success] = "CheckIn at #{time.to_formatted_s(:short)}. Have a nice day"
+    end
+    redirect_to action: :new
+  rescue StandardError => e
+    flash[:danger] = 'Private number not exist. Please try again'
+    redirect_to action: :new
+  end
```
this logic is a bit hard to understand. I would move it to the model where it belongs and keep the controller skinny, quick example => https://riptutorial.com/ruby-on-rails/example/9609/fat-model--skinny-controller

Javi1499 replied:
Thanks. I changed my code using this format

femm34 committed:
bootstrap alert when the user creates a new branch

i used .present? method to know if notice had something and if was true showed the alert, also enabled the function to show the branches form

femm34 committed:
Merge branch 'master' into 21-main-view-branch-section

femm34 committed:
merge branch master into main view main section branch

Javi1499 committed:
Fat model, skinny controller implemented

Javi1499 committed:
Fat model, skinny controller implemented

femm34 committed:
main view of branches finished

i implemented js to show a form and also hide a button, i changed some styles

femm34 committed:
i modified the spaces  in all .html.erb files

YilKanda commented on app/helpers/records_helper.rb:
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+module RecordsHelper
+
+end
```
If you don't use this helper, you can remove it.

femm34 committed:
cleaning file

i removed unused code

Javi1499 committed:
Record herlper was deleted

Javi1499 committed:
Record herlper was deleted

femm34 committed:
Merge pull request #25 from BC-MAY-21-ROR/21-main-view-branch-section

21 main view branch section

Javi1499 committed:
Typo error was corrected and rebase master

Javi1499 committed:
Merge branch '18-creation-of-main-view' of https://github.com/BC-MAY-21-ROR/kata-07-reloj-checador-k7-f22-t1 into 18-creation-of-main-view

femm34 reviewed (approved):
LGTM :)

germanruzca reviewed (approved):
LGTM

EliasUR reviewed (approved):
LGTM
